### ISSUE TITLE
Multil-speaker

### ISSUE THREAD

**Issue by TuananhCR** [stlohrey/dia-finetuning#14]
Hi I wondering about the data training, it does have channel_name , .wav and .text. Is the code finetune is training for conversation between 2 people, because I can't do like [S1] and [S2] stuffs like in Dia.

**Comment by alisson-anjos** [comment]
I trained a V2 of my model in Portuguese, and in it I used audio examples from podcasts or newspapers from Brazil in my dataset, that is, in the transcriptions I used Gemini to mark the speakers and non-verbal effects, I did the test after training and it can generate more than one voice per conversation, the big issue is that the dataset I assembled forgot to remove the background noise from the audio and also that there was a problem in the training script that did not add the ending token and with that all the audios that I generate, even if it is a short transcription, it generates a 30-second audio, but here is an example from my dataset.

![Image](https://github.com/user-attachments/assets/1985f46e-7e55-4afc-b038-a5b10155bbbf)

**Comment by TuananhCR** [comment]
> I trained a V2 of my model in Portuguese, and in it I used audio examples from podcasts or newspapers from Brazil in my dataset, that is, in the transcriptions I used Gemini to mark the speakers and non-verbal effects, I did the test after training and it can generate more than one voice per conversation, the big issue is that the dataset I assembled forgot to remove the background noise from the audio and also that there was a problem in the training script that did not add the ending token and with that all the audios that I generate, even if it is a short transcription, it generates a 30-second audio, but here is an example from my dataset.
> 
> ![Image](https://github.com/user-attachments/assets/1985f46e-7e55-4afc-b038-a5b10155bbbf)

So did you mean in text wav file do we need it to be conversastion and [S1] and [S2] and (emotion) in advanced ? Can we assign each speaker is their own channel name like [mrbeast] , and without (emotion) in text at first. Can I train it and resulted it emotion and conversation ?
Thank you for answer my concerns

**Comment by Respaired** [comment]
> I trained a V2 of my model in Portuguese, and in it I used audio examples from podcasts or newspapers from Brazil in my dataset, that is, in the transcriptions I used Gemini to mark the speakers and non-verbal effects, I did the test after training and it can generate more than one voice per conversation, the big issue is that the dataset I assembled forgot to remove the background noise from the audio and also that there was a problem in the training script that did not add the ending token and with that all the audios that I generate, even if it is a short transcription, it generates a 30-second audio, but here is an example from my dataset.
> 
> ![Image](https://github.com/user-attachments/assets/1985f46e-7e55-4afc-b038-a5b10155bbbf)

@alisson-anjos 
Sorry to tag you, but Have you had any success with fixing the issue with generating an audio that's too long? 

it constantly goes beyond the 3072 duation I set for training.

**Comment by alisson-anjos** [comment]
> > I trained a V2 of my model in Portuguese, and in it I used audio examples from podcasts or newspapers from Brazil in my dataset, that is, in the transcriptions I used Gemini to mark the speakers and non-verbal effects, I did the test after training and it can generate more than one voice per conversation, the big issue is that the dataset I assembled forgot to remove the background noise from the audio and also that there was a problem in the training script that did not add the ending token and with that all the audios that I generate, even if it is a short transcription, it generates a 30-second audio, but here is an example from my dataset.
> > ![Image](https://github.com/user-attachments/assets/1985f46e-7e55-4afc-b038-a5b10155bbbf)
> 
> [@alisson-anjos](https://github.com/alisson-anjos) Sorry to tag you, but Have you had any success with fixing the issue with generating an audio that's too long?
> 
> it constantly goes beyond the 3072 duation I set for training.

Did you use Stlohrey version of the script? He told me he had fixed it, that there was a bug due to the missing stop token. I haven't retrained the V2 version of my model yet, but I'll do it in the next few days, I'll build a huge dataset using gemini and etc and I can't abandon V2 now.

**Comment by Respaired** [comment]
> > > I trained a V2 of my model in Portuguese, and in it I used audio examples from podcasts or newspapers from Brazil in my dataset, that is, in the transcriptions I used Gemini to mark the speakers and non-verbal effects, I did the test after training and it can generate more than one voice per conversation, the big issue is that the dataset I assembled forgot to remove the background noise from the audio and also that there was a problem in the training script that did not add the ending token and with that all the audios that I generate, even if it is a short transcription, it generates a 30-second audio, but here is an example from my dataset.
> > > ![Image](https://github.com/user-attachments/assets/1985f46e-7e55-4afc-b038-a5b10155bbbf)
> > 
> > 
> > [@alisson-anjos](https://github.com/alisson-anjos) Sorry to tag you, but Have you had any success with fixing the issue with generating an audio that's too long?
> > it constantly goes beyond the 3072 duation I set for training.
> 
> Did you use Sthlorey's version of the script? He told me he had fixed it, that there was a bug due to the missing stop token. I haven't retrained the V2 version of my model yet, but I'll do it in the next few days, I'll build a huge dataset using gemini and etc and I can't abandon V2 now.

That's what i'm using, i also modified it a little bit; because It appears the shifted BOS tokens are not masked in this version as well, so the BOS is learnt for the delay slots. I'm not sure this is the intended behavior. (highly doubt)

@stlohrey

**Comment by stlohrey** [comment]
@Respaired BOS is injected after the delay pattern is applied, so it should not be shifted and should be sliced correctly before loss computation.

**Comment by TuananhCR** [comment]
> I trained a V2 of my model in Portuguese, and in it I used audio examples from podcasts or newspapers from Brazil in my dataset, that is, in the transcriptions I used Gemini to mark the speakers and non-verbal effects, I did the test after training and it can generate more than one voice per conversation, the big issue is that the dataset I assembled forgot to remove the background noise from the audio and also that there was a problem in the training script that did not add the ending token and with that all the audios that I generate, even if it is a short transcription, it generates a 30-second audio, but here is an example from my dataset.
> 
> ![Image](https://github.com/user-attachments/assets/1985f46e-7e55-4afc-b038-a5b10155bbbf)

Can I ask which Gemini model did you used to do the transcript ?

**Comment by alisson-anjos** [comment]
> > I trained a V2 of my model in Portuguese, and in it I used audio examples from podcasts or newspapers from Brazil in my dataset, that is, in the transcriptions I used Gemini to mark the speakers and non-verbal effects, I did the test after training and it can generate more than one voice per conversation, the big issue is that the dataset I assembled forgot to remove the background noise from the audio and also that there was a problem in the training script that did not add the ending token and with that all the audios that I generate, even if it is a short transcription, it generates a 30-second audio, but here is an example from my dataset.
> > ![Image](https://github.com/user-attachments/assets/1985f46e-7e55-4afc-b038-a5b10155bbbf)
> 
> Can I ask which Gemini model did you used to do the transcript ?

gemini-2.5-flash
